### PR TITLE
update docstr

### PR DESCRIPTION
**Pull request by sonichi**
Update docstr for better documentation webpage.
@qingyun-wu Please review. The online automl part requires more work. Please complete them. Refer to the auto-generated webpage for formatting issues.

### CODE REVIEW

**Comment by int-chaos** [comment]
```
    @property
    def search_space(self) -> dict:
        """Search space
        Must be called after fit(...) (use max_iter=0 to prevent actual fitting)
        Returns:
            A dict of the search space
        """
```

automl.py lines 1078-1085; should there be a period at the end of `A dict of the search space`

**Comment by int-chaos** [comment]
image

periods at the end of each

**Comment by int-chaos** [comment]
automl lines 1112 and 1146

delete lines

File: flaml/onlineml/autovw.py [hunk 1]
```diff
@@ -85,100 +96,121 @@ def __init__(self,
...
+            data_sample (vw_example): one data sample on which the model gets updated.
         """
         self._iter += 1
         self._trial_runner.step(data_sample, (self._y_predict, self._best_trial))
 
     def _select_best_trial(self):
-        """Select a best trial from the running trials accoring to the _model_select_policy
-        """
-        best_score = float('+inf') if self._model_selection_mode == 'min' else float('-inf')
+        """Select a best trial from the running trials accoring to the _model_select_policy"""
```
Comment (int-chaos) [inline flaml/onlineml/autovw.py:172]: `"""Select a best trial from the running trials accoring to the _model_select_policy."""`

File: flaml/onlineml/trial.py [hunk 2]
```diff
@@ -85,32 +92,40 @@ def __init__(self, result_type_name: str, cb_coef: Optional[float] = None,
...
 
-    def _update_loss_cb(self, bound_of_range, data_dim,
-                        bound_name='sample_complexity_bound'):
-        """Calculate bound coef
-        """
-        if bound_name == 'sample_complexity_bound':
+    def _update_loss_cb(
+        self, bound_of_range, data_dim, bound_name="sample_complexity_bound"
+    ):
+        """Calculate bound coef"""
```
Comment (int-chaos) [inline flaml/onlineml/trial.py:116]: `"""Calculate bound coef."""`

File: flaml/onlineml/trial.py [hunk 3]
```diff
@@ -320,45 +345,48 @@ def __init__(self,
 
     @staticmethod
     def _config_to_id(config):
-        """Generate an id for the provided config
-        """
+        """Generate an id for the provided config"""
```
Comment (int-chaos) [inline flaml/onlineml/trial.py:348]: `"""Generate an id for the provided config."""`

File: flaml/onlineml/trial.py [hunk 4]
```diff
@@ -369,53 +397,59 @@ def train_eval_model_online(self, data_sample, y_pred):
...
                 # clip y_pred in the observed range of y
                 y_pred = min(y_max_observed, max(y_pred, y_min_observed))
         else:
             raise NotImplementedError
         return loss_func([y_true], [y_pred])
 
     def _update_y_range(self, y):
-        """Maintain running observed minimum and maximum target value
-        """
+        """Maintain running observed minimum and maximum target value"""
```
Comment (int-chaos) [inline flaml/onlineml/trial.py:442]: `"""Maintain running observed minimum and maximum target value."""`

File: flaml/onlineml/trial.py [hunk 5]
```diff
@@ -369,53 +397,59 @@ def train_eval_model_online(self, data_sample, y_pred):
...
             self._y_max_observed = y
 
     @staticmethod
-    def _get_dim_from_ns(namespace_feature_dim: dict, namespace_interactions: [set, list]):
-        """Get the dimensionality of the corresponding feature of input namespace set
-        """
+    def _get_dim_from_ns(
+        namespace_feature_dim: dict, namespace_interactions: [set, list]
+    ):
+        """Get the dimensionality of the corresponding feature of input namespace set"""
```
Comment (int-chaos) [inline flaml/onlineml/trial.py:452]: `"""Get the dimensionality of the corresponding feature of input namespace set."""`

File: flaml/scheduler/online_scheduler.py [hunk 6]
```diff
@@ -48,62 +53,68 @@ class OnlineSuccessiveDoublingScheduler(OnlineScheduler):
...
+                trial.trial_id,
+                trial.result.resource_used,
+                trial.resource_lease,
+            )
             return TrialScheduler.PAUSE
 
 
 class ChaChaScheduler(OnlineSuccessiveDoublingScheduler):
-    """  Keep the top performing learners running
+    """Keep the top performing learners running
```
Comment (int-chaos) [inline flaml/scheduler/online_scheduler.py:92]: `"""Keep the top performing learners running.`

File: flaml/data.py [hunk 7]
```diff
@@ -208,9 +205,21 @@ def concat(X1, X2):
 
 
 class DataTransformer:
-    """transform X, y"""
+    """Transform input training data"""
```
Comment (int-chaos) [inline flaml/data.py:208]: `"""Transform input training data."""`

**Review by sonichi — COMMENTED** [review]
Looks good.

**Review by int-chaos — APPROVED** [review]
looks good, awaiting another reviewer's approval.

**Pull request merged**